### PR TITLE
Email verification

### PR DESCRIPTION
**Pull request by t-romani**
### Jira Board Card
[As a user I should be able to confirm my account after signing up in order to ensure that all the user's email are valid](https://trello.com/c/wmCa5NSD/16-3-as-a-user-i-should-be-able-to-confirm-my-account-after-signing-up-in-order-to-ensure-that-all-the-users-email-are-valid)

🧪 Includes test on email sending
🦞 Includes test on login without email onfirmation & updates on other login tests.

### CODE REVIEW

File: target_mvd/settings.py [hunk 1]
```diff
@@ -152,14 +155,25 @@
...
     'django.contrib.auth.backends.ModelBackend',
 ]
 
+# email settings
+EMAIL_BACKEND = env('EMAIL_BACKEND')
+EMAIL_HOST = env('EMAIL_HOST')
+EMAIL_USE_TLS = True
+EMAIL_PORT = env('EMAIL_PORT')
+EMAIL_HOST_USER = env('EMAIL_HOST_USER')
+EMAIL_HOST_PASSWORD = env('EMAIL_HOST_PASSWORD')
```
Comment (nsantos16) [inline target_mvd/settings.py]: Maybe add documentation on what the environment variables are, WDYT?

File: users/tests/test_user_signup.py [hunk 2]
```diff
@@ -42,3 +44,11 @@ def test_sign_up_email_already_exists(self):
 
         self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
         self.assertEqual(response.data['email'], [SignUpTests.ALREADY_REGISTERED])
+
+    def test_confirmation_email_sent(self):
+        self.data.update({'password1': self.password, 'password2': self.password})
+        self.client.post('/dj-rest-auth/registration/', self.data)
```
Comment (nsantos16) [inline users/tests/test_user_signup.py]: You can use [reverse](https://docs.djangoproject.com/en/3.1/ref/urlresolvers/) here, in this way if the URL of the endpoint change you don't have to change in all the places it has been used

File: target_mvd/urls.py [hunk 3]
```diff
@@ -13,11 +13,14 @@
     1. Import the include() function: from django.urls import include, path
     2. Add a URL to urlpatterns:  path('blog/', include('blog.urls'))
 """
+from dj_rest_auth.registration.views import ConfirmEmailView, VerifyEmailView
 from django.contrib import admin
 from django.urls import include, path
 
 urlpatterns = [
     path('admin/', admin.site.urls),
+    path('dj-rest-auth/registration/account-confirm-email/<str:key>/',ConfirmEmailView.as_view(),),
```
Comment (nsantos16) [inline target_mvd/urls.py]: WDYT about changing a little bit the name of the endpoints? Like `registration/account-confirm-email/<str:key>/` for example

File: users/tests/test_user_signup.py [hunk 4]
```diff
@@ -42,3 +44,11 @@ def test_sign_up_email_already_exists(self):
...
+
+        self.assertEqual(len(mail.outbox), 1)
+        self.assertEqual(mail.outbox[0].to, [User.objects.get().email])
+        self.assertEqual(mail.outbox[0].subject, f"[{SignUpTests.SITE_NAME}] Please Confirm Your E-mail Address")
```
Comment (brunomichetti) [inline users/tests/test_user_signup.py]: Please use single quotes here

File: users/tests/test_user_signup.py [hunk 5]
```diff
@@ -10,6 +11,7 @@
 class SignUpTests(APITestCase):
```
Comment (brunomichetti) [inline users/tests/test_user_signup.py:13]: You should add a unit test to check this:
a new user singup
check that the user has been created in the db but he can't login
confirm the account
check that the user can login

File: users/tests/test_user_signup.py (hunk 2 above)
Comment (brunomichetti) [reply users/tests/test_user_signup.py]: Yes, just as `self.sign_in_url`

**Review by brunomichetti — CHANGES_REQUESTED** [review]

**Review by brunomichetti — APPROVED** [review]

File: users/tests/test_user_login_logout.py [hunk 6]
```diff
@@ -40,11 +43,21 @@ def test_sign_in_unexistant_email(self):
...
+        response = self.client.post(reverse('rest_logout'))
 
         self.assertEqual(response.status_code, status.HTTP_200_OK)
         self.assertTrue('detail' in response.data)
 
+    def test_login_without_confirmation(self):
+        ea = EmailAddress.objects.get()
+        ea.verified = False
+        ea.save()
+        data = {'email': self.user.email, 'password': self.password}
```
Comment (brunomichetti) [inline users/tests/test_user_login_logout.py:55]: Just a tip, maybe this is not actually the case:
If you want to get a dict from a factory object, you can do

`data_dict = factory.build(dict, FACTORY_CLASS=MyFactoryClass)`

**Review by nsantos16 — APPROVED** [review]

**Pull request merged**